packages/react-aria-components/test/AriaAutocomplete.test-util.tsx

```diff
@@ -0,0 +1,572 @@
+        expect(input.selectionStart).toBe(2);
+
+        await user.keyboard('{ArrowDown}');
+        act(() => jest.runAllTimers());
+        expect(input.selectionStart).toBe(2);
+
+        await user.keyboard('{ArrowUp}');
+        act(() => jest.runAllTimers());
+        expect(input.selectionStart).toBe(2);
+      });
+
+      if (ariaPattern === 'menu') {
+        it('should update the aria-activedescendant when hovering over an item', async function () {
+          let {getByRole} = renderers.standard();
+          let input = getByRole('searchbox');
+          let menu = getByRole(collectionNodeRole);
+          let options = within(menu).getAllByRole(collectionItemRole);
+          expect(input).not.toHaveAttribute('aria-activedescendant');
+
+          await user.tab();
+          expect(document.activeElement).toBe(input);
+          // Need to press to set a modality
+          await user.click(input);
+          await user.hover(options[1]);
+          act(() => jest.runAllTimers());
+          expect(input).toHaveAttribute('aria-activedescendant', options[1].id);
+          expect(document.activeElement).toBe(input);
+        });
+      }
+    });
+
+    if (renderers.defaultValue) {
+      describe('default text value', function () {
+        it('should support default value', async function () {
+          let {getByRole} = (renderers.defaultValue!)();
+          let input = getByRole('searchbox');
+          expect(input).toHaveValue('Ba');
+          let menu = getByRole(collectionNodeRole);
+          let options = within(menu).getAllByRole(collectionItemRole);
+          expect(options).toHaveLength(2);
+          expect(options[0]).toHaveTextContent('Bar');
+          expect(options[1]).toHaveTextContent('Baz');
+
+          await user.tab();
+          expect(document.activeElement).toBe(input);
+          await user.keyboard('z');
+          act(() => jest.runAllTimers());
+          options = within(menu).getAllByRole(collectionItemRole);
+          expect(options).toHaveLength(1);
+          expect(options[0]).toHaveTextContent('Baz');
+        });
+      });
+    }
+
+    if (ariaPattern === 'menu' && renderers.itemActions) {
+      describe('item actions', function () {
+        it('should trigger the wrapped element\'s actionListener when hitting Enter', async function () {
+          let {getByRole} = (renderers.itemActions!)();
+          let input = getByRole('searchbox');
+          let menu = getByRole(collectionNodeRole);
+          expect(input).not.toHaveAttribute('aria-activedescendant');
+
+          await user.tab();
+          expect(document.activeElement).toBe(input);
+
+          await user.keyboard('{ArrowDown}');
+          let options = within(menu).getAllByRole(collectionItemRole);
+          expect(input).toHaveAttribute('aria-activedescendant', options[0].id);
+          await user.keyboard('{Enter}');
+          expect(actionListener).toHaveBeenCalledTimes(1);
```

Comment:
better, however, this is still tied to our API.
Someone could implement an autocomplete that is self contained and fires no event (I'm not sure why, but they could, maybe they fire a custom event on the dom instead of using props)

In addition, they may not include the id of the item, maybe they've created some mapping of the ids to some database uuid or something.

Reply:
The `actionListener` here that they provided doesn't have to be passed to component via props though, they just need to provide us with whatever action listener mock that would get triggered on item action and we assert it has been properly called when hitting enter on the input while that item is focused right?

As for the id of the item, the id being checked above is the `aria-activedescendant` which needs to be tied to the dom node they are virtually focusing so that should be fine as well right?